### ISSUE TITLE
UnitTests: two test Failures in Compile.t

### ISSUE THREAD

**Issue by bschmalhofer** [RotherOSS/otobo#486]
The cause is that ``Kernel::System::DateTime`` is loaded for ``use DateTime 1.08``, which is not wanted.
The wrong module is loaded because _Kernel/System_ is in ``@INC``. The reason for that is not obvious. It is probably some unfortunate ``use lib $FindBin::Bin`` statement.
```
 DateTime defines neither package nor VERSION--version check failed at /opt/otobo/Kernel/System/DateTime.pm line 38.
# BEGIN failed--compilation aborted at /opt/otobo/Kernel/System/DateTime.pm line 38.
# Compilation failed in require at /opt/otobo/Kernel/System/DateTime.pm line 38.
# BEGIN failed--compilation aborted at /opt/otobo/Kernel/System/DateTime.pm line 38.
# Compilation failed in require at /opt/otobo/Kernel/Language.pm line 28.
# BEGIN failed--compilation aborted at /opt/otobo/Kernel/Language.pm line 28.
# Compilation failed in require at /opt/otobo/Kernel/Output/HTML/Layout.pm line 30.
# BEGIN failed--compilation aborted at /opt/otobo/Kernel/Output/HTML/Layout.pm line 30.
# Compilation failed in require at /opt/otobo/Kernel/System/ObjectManager.pm line 34.
# BEGIN failed--compilation aborted at /opt/otobo/Kernel/System/ObjectManager.pm line 34.
# Compilation failed in require at Kernel/System/UnitTest/RegisterDriver.pm line 42.
# BEGIN failed--compilation aborted at Kernel/System/UnitTest/RegisterDriver.pm line 42.

```

**Comment by bschmalhofer** [comment]
`Kernel::Config::Defaults` is the culprit. This module  sets `@INC` based on the location of the including script. A minimal example is:

```
otobo@5bc1685c346c:~$ cat Kernel/System/UnitTest/t.pl 
use FindBin qw($Bin);
use lib "$Bin/..";

use DateTime 1.08;
otobo@5bc1685c346c:~$ perl -c Kernel/System/UnitTest/t.pl 
DateTime defines neither package nor VERSION--version check failed at /opt/otobo/Kernel/System/UnitTest/../DateTime.pm line 38.
BEGIN failed--compilation aborted at /opt/otobo/Kernel/System/UnitTest/../DateTime.pm line 38.
Compilation failed in require at Kernel/System/UnitTest/t.pl line 4.
BEGIN failed--compilation aborted at Kernel/System/UnitTest/t.pl line 4.
otobo@5bc1685c346c:~$ 
```

_Kernel::System_ is added to `@INC`, thus` Kernel::System::DateTime `is loaded, which has no version.

The fix is to remove the setup of `@INC` in _Kernel/Config/Defaults.pm_.

This was already fixed in #174, https://github.com/RotherOSS/otobo/commit/cc36ed8c2867af50d6ffa0854e565ca8129dd91d, but the patch was reverted because apparently other problems propped up. For OTOBO 10.1 this was fixed in #222.

**Comment by bschmalhofer** [comment]
Tests look fine. Closing the issue.
